### DISCUSSION TITLE
Next Build

### DISCUSSION THREAD

**Discussion by DaxSoft** [vercel/next.js#17027] [Help]
**My doubt is the following one**:
  If my webapp is on production mode (online), and I need to commit a new change and by this, it will requires me to run the command 'next build'. My question is, if everytime that I hit 'next build', I would need to restart the webapp? 

**The origin of my doubt is:**
 Let's suppose that I have made a webapp using Next with a custom server. And let's imagine that my webapp has a 'Function', or 'Method', that somehow receives a new page to the folder **page**, with a brand new component that my original webapp doesn't have. So, by the all means, I would need to run 'next build' to make it effective. However, if I do it, my concern is if it will not 'break/crash' the on-going webapp that it's online.
 Sure thing I can use the '/page/[:slug]', however, it's only effective if my webapp has all components/code for it. The origin point is that the 'Function/Method (mentioned previous)', will receive a new file for the folder page (doing it from the server, of course), with a code (component) that my webapp don't have originally.

**Comment by leerob** [comment]
I think you might be confusing two separate things. 

- `next build` creates an optimized production build of your application.
- `next start` starts the application in production mode. You have to run `next build` first.

https://nextjs.org/docs/api-reference/cli

**Reply by DaxSoft** [reply]
Yes, I know the difference. However is the main function of the 'next build', that I'm trying to pick-up. I mean, my webapp will receive a brand new file on /page. And I want to generate a new optimized production in which, this new file goes optimized as well

**Reply by leerob** [reply]
Are you asking what happens if you use dynamic routing (e.g. `pages/[slug].js`) and the requested slug (e.g. `my-new-slug`) is not available?

You are correct that one option is to do a full rebuild your application with `next build` However, you can also use [fallback](https://nextjs.org/docs/basic-features/data-fetching#fallback-false) to fetch your new date (`my-new-slug`) on the fly. For example, if you were building a blog you would want to fetch the new post (`my-new-slug`). Here's some example pseudo-code:

```js
export default function BlogPost({ post }) {
  const router = useRouter()
  if (!router.isFallback && !post?.slug) {
    return <ErrorPage statusCode={404} />
  }

  return router.isFallback ? (
      <PostTitle>Loading…</PostTitle>
    ) : (
      <BlogPost />
  );
}

export async function getStaticProps({ params }) {
  const post = await getPost(params.slug)

  return {
    props: {
      post
    },
  }
}

export async function getStaticPaths() {
  const posts = getAllPosts(['slug'])

  return {
    paths: posts.map((post) => {
      return {
        params: {
          slug: post.slug,
        },
      }
    }),
    fallback: false,
  }
}
```

And a [real example](https://github.com/vercel/next.js/blob/canary/examples/blog-starter/pages/posts/%5Bslug%5D.js).

**Reply by DaxSoft** [reply]
No, no this case. Let me try to explain in another form;
Suppose that my structure is this:
 **server/**
   _index.ts_
   **server/routes/**
     _function.ts_

**page/**
  _index.tsx_
  **page/test/** 
   _a.tsx (uses comp-a.tsx)_ 

**client/**
   **client/components/**
     _comp-a.tsx_

Then, let's imagine that in the route 'function.ts', has the method to get a brand new file to the 'page folder'. So my page folder will receive another file on the 'folder' 'test/'; as well, on 'client'/component', will recever another file;
**page/**
  index.tsx
  **page/test/** 
   _a.tsx (uses comp-a.tsx) 
   b.tsx (usex comp-b.tsx)_
**client/**
   **client/components/**
     _comp-a.tsx
     comp-b.tsx_

To get a optimized bundle, I would need to run the 'next-build'. My doubt is: if I do it, my webapp in produciton will crash/break and would need to restart?

**Reply by leerob** [reply]
I'm not gonna lie, I'm pretty confused 😆  Are you trying to dynamically create routes on the fly with a custom server? What is your use case - what are you trying to build?

**Reply by DaxSoft** [reply]
Use case: let's imagina that I have something like 'https://linktr.ee/', pretty sure they use the [slug], to their dynamic routes. However, it can be happen because they use "standard" options to customize the dynamic page. So the user itself, can just handle with their options and create a custom page the way he's like

Now, let's imagine the same case, however now, I receive a client that needs an totally new page that has new functions and it will require me to create a new code to create the page that my client needs. In this case, the [slug] it will not be useful, because my original webapp, doesn't have the correct component to handle the custom page. I would need to create a new file with the component, to the new page.

**Reply by leerob** [reply]
Could you share any code or any example?

**Reply by DaxSoft** [reply]
Unfortunally, no, I don't have any code at moment, sure thing I can create one, but only at sunday (I work hard tomorrow). It's more a concept. 

Sure, we can imagine this as 'DLC for game' or 'plugin for system', however it's more like 'page + component for website'.
